lib/qmp.ml

```diff
@@ -150,6 +170,12 @@ let message_of_string x =
           let service = int_of_string (string (List.assoc "service" list)) in
           let host = string (List.assoc "host" list) in
           {enabled; auth; family; service; host})
+      | `Assoc list when List.mem_assoc "product-num" list
+                      && List.mem_assoc "build-num" list ->
+        Xen_platform_pv_driver_info (
+          let product_num = int (List.assoc "product-num" list) in
```

Comment:
That would return a 'Failure("int")' exception. I've added a new patch 865c4af below that causes the library to return an Error message if the parsing of the json parameters fail, and add this as a new unit test case as discussed.